File: src/main/java/me/arrush/javabase/databases/MariaDatabase.java
package me.arrush.javabase.databases;

import java.sql.SQLException;

@SuppressWarnings({"WeakerAccess", "unused"})
public final class MariaDatabase extends Database {
    MariaDatabase(String hostName, String dbName, boolean usePipelineAuth) throws ClassNotFoundException, SQLException {
        super(dbName, DatabaseType.MARIADB, "org.mariadb.jdbc.Driver", getMariaConnection(hostName, dbName, usePipelineAuth));
    }

    MariaDatabase(String hostName, String dbName) throws SQLException, ClassNotFoundException {
        this(hostName, dbName, false);
    }
}

File: src/main/java/com/arrushc/javabase/entities/database/MariaDatabase.kt
package com.arrushc.javabase.entities.database

import com.arrushc.javabase.entities.Database

class MariaDatabase internal constructor(hostName: String?, dbName: String?, username: String, password: String, usePipelineAuth: Boolean = false) : Database(dbName!!, DatabaseType.MARIADB, "org.mariadb.jdbc.Driver", getMariaDataSource(hostName, dbName, username, password, usePipelineAuth))